src/parser.cc

```diff
@@ -0,0 +1,90 @@
+#include "xcshell/parser.h"
+
+#include <string>
+#include <vector>
+
+#include "xcshell/redirectelement.h"
+#include "xcshell/utils.h"
+
+std::vector<std::string> getArgs(std::vector<std::string> &command_with_args) {
```

Comment:
Could `command_with_args ` be const reference here?

Reply:
Can this method become private member of Parser class? Instead of let it become a global function?